include/toml++/impl/source_region.hpp

```diff
@@ -217,6 +217,70 @@ TOML_NAMESPACE_START
 			return lhs;
 		}
 	};
+
+	/// \brief	Returns the line at the specified line number, from the specified document.
+	///
+	/// \detail \cpp
+	/// auto doc = "alpha = 1\nbeta = 2"sv;
+	/// auto second_line = toml::get_line(doc, 2);
+	///
+	/// std::cout << "The second line says \"" << second_line << "\"\n";
+	/// \ecpp
+	///
+	/// \out
+	/// The second line says "beta = 2"
+	/// \eout
+	///
+	/// \param 	doc	The document.
+	/// \param 	line_num	The line number (1-based).
+	///
+	/// \returns	The specified line, excluding any possible trailing carriage return or line feed character.
+	/// \remarks Returns an empty string_view when there is no line at the specified line number, in the specified document.
+	inline std::string_view get_line(std::string_view doc, source_index line_num)
```

Comment:
Looks good to me, only thing I'd want to address before accepting this contribution would be the `inline`-ness of it. The library supports traditional compilation as static/shared - this function being a regular old non-template free function makes it a perfect canditate to be moved to the "source" section.

Because `toml++` is also header-only by default, the way this works is a bit non-obvious. The changes you'd need to make would consist of three main steps:

1. Change this to a declaration with the right decorators, like:
```suggestion
	TOML_NODISCARD
	TOML_EXPORTED_FREE_FUNCTION
	std::string_view get_line(std::string_view doc, source_index line_num);
```

2. Create an ".inl" file to host the implementation (see `impl/at_path.inl` for implementation guidance)
3. Add an include for the implementation part in toml.hpp ([examples](https://github.com/marzer/tomlplusplus/blob/cbc00d6766f14bcd2b3e72de2a3c369a0d48b98a/include/toml%2B%2B/toml.hpp#L60-L76))

Of course I can do these parts for you too, if you like.

Reply:
Hi Mark, I just addressed your comment on `\r`. (Please check, as: [force-pushed](https://github.com/marzer/tomlplusplus/compare/61c233b50cae0826cf7724b2ace201e4f57c2725..81bff73982063fbc35698e5a17ee076af8bc1967)) Can you please take care of the ".inl" part?

Reply:
Yep, sure.

Reply:
Thanks Mark! By the way, I personally don't use `noexcept` so extensively, which is why I originally forgot to add one here. I became more reluctant to use it after seeing some disappointing performance effects from adding `noexcept`, from my [noexcept_benchmark](https://github.com/N-Dekker/noexcept_benchmark). Only for move constructors, move assignments, and `swap` functions, the benefit of `noexcept` looks obvious to me. But then again, marking get_line  `noexcept` is no problem to me, it won't harm 😃

Reply:
I do not think of `noexcept` as a performance hint, but a documentation tool. If the function doesn't throw, I mark it as `noexcept` to indicate as much.

Reply:
Thanks Mark, I see your point. On the other hand, adding a `noexcept` specifier may also have the effect of causing a function to terminate, instead of throwing an exception. I think I often prefer an exception over an instant program termination. 🤷

Reply:
By the way, technically it appears possible to declare `get_line` constexpr. Personally I don't need that right now, but maybe it's still something for you to consider...?

Reply:
Thanks Mark, I generally agree that a function that may leak exceptions should not be marked `noexcept`. (Except for _maybe_ a move constructor or move assignment operator.) So honestly I feel a bit unsure about C++ Core Guidelines item [Use `noexcept` when exiting a function because of a `throw` is impossible or unacceptable](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Re-noexcept) Which says:

> _Example_
```
vector<double> munge(const vector<double>& v) noexcept
{
    vector<double> v2(v.size());
    // ... do something ...
}
```
> The noexcept here states that I am not willing or able to handle the situation where I cannot construct the local vector. That is, I consider memory exhaustion a serious design error (on par with hardware failures) so that I’m willing to crash the program if it happens.

In general, I would not mark a function that might create a non-empty `std::vector` noexcept, would you?